src/components/Button/StatusButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { GestureResponderEvent, Pressable, StyleSheet } from 'react-native';
+
+import { selectStatus, useTheme } from '@/hooks';
+import { Status } from '@/types/status';
+
+type Props = React.ComponentPropsWithRef<typeof Pressable> & {
+  status: Status;
+  onPress: (event: GestureResponderEvent) => void;
+};
+
+const StatusButton = (param: Props) => {
+  const { themedStyles } = useTheme();
+  const actualStatus: Status = 'Inaccessible'; //added temporarily until the selected status is supported
+  const buttonStatus: Status = param.status;
```

Comment:
nie musisz przypisywać tego do zmiennej jeśli i tak wykorzystujesz to w kodzie jeden raz